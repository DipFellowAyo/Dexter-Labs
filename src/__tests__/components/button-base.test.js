import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import ButtonBase from '../../components/buttons/button-base'

describe('ButtonBase Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <ButtonBase variant='contained' color='primary' text='Click me' />
    )
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('accept a variant as props', () => {
    expect(wrapper.prop('variant')).toEqual('contained')
  })

  it('accept a color as props', () => {
    expect(wrapper.prop('color')).toEqual('primary')
  })

  it('accept a disableElevation as props', () => {
    expect(wrapper.prop('disableElevation')).toEqual(true)
  })

  it('accept a text as children props', () => {
    expect(wrapper.prop('children')).toEqual('Click me')
  })
})
