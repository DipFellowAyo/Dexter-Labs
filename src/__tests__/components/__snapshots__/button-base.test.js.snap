// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBase Component should match snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-buttonStyles-1"
  color="primary"
  disableElevation={true}
  variant="contained"
>
  Click me
</WithStyles(ForwardRef(Button))>
`;
